properties/basic.py

```diff
@@ -110,7 +131,7 @@ def assert_valid(self, instance):
                 )
             )
         if value is not None:
-            self.validate(instance, value)
+            instance._backend[self.name] = self.validate(instance, value)
```

Comment:
I think this covers the unlikely edge case of something like:
```
class HasColor(properties.HasProperties):
    c = properties.Color('just a color')

hc = HasColor()
hc._backend['c'] = 'red'
hc.validate()
```
Either way, `hc.validate()` is `True`, but without setting on `assert_valid`, `hc.c` is still `'red'` rather than `[255, 0, 0]`.

I wasn't really sure about this change. I don't think we need to be exceptionally accommodating to direct modifications of `_backend` (change notifications already won't fire, and it is a hidden variable after all), but this just adds one more validation pass...